it/src/main/java/org/corfudb/universe/node/server/docker/DockerCorfuServer.java

```diff
@@ -0,0 +1,403 @@
+            execCommand("iptables", "-A", "INPUT", "-s", gateway, "-j", "ACCEPT");
+            // iptables -A INPUT -s $subnet -j DROP
+            execCommand("iptables", "-A", "INPUT", "-s", subnet, "-j", "DROP");
+            // iptables -A OUTPUT -d $gateway -j ACCEPT
+            execCommand("iptables", "-A", "OUTPUT", "-d", gateway, "-j", "ACCEPT");
+            // iptables -A OUTPUT -d $subnet -j DROP
+            execCommand("iptables", "-A", "OUTPUT", "-d", subnet, "-j", "DROP");
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't disconnect container from docker network " + params.getName(), ex);
+        }
+    }
+
+    /**
+     * Pause the container from docker network
+     *
+     * @throws NodeException this exception will be thrown if the server can not be paused
+     */
+    @Override
+    public void pause() {
+        log.info("Pausing the Corfu server: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (!container.state().running()) {
+                log.warn("The container `{}` is not running", container.name());
+                return;
+            }
+            docker.pauseContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't pause container " + params.getName(), ex);
+        }
+    }
+
+    /**
+     * Start a {@link Node}
+     *
+     * @throws NodeException this exception will be thrown if the server can not be started
+     */
+    @Override
+    public void start() {
+        log.info("Starting the corfu server: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (container.state().running() || container.state().paused()) {
+                log.warn("The container `{}` already running, should stop before start", container.name());
+                return;
+            }
+            docker.startContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't start container " + params.getName(), ex);
+        }
+    }
+
+    /**
+     * Restart a {@link Node}
+     *
+     * @throws NodeException this exception will be thrown if the server can not be restarted
+     */
+    @Override
+    public void restart() {
+        log.info("Restarting the corfu server: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (container.state().running() || container.state().paused()) {
+                log.warn("The container `{}` already running, should stop before restart", container.name());
+                return;
+            }
+            docker.restartContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)